examples/notion-github-sync/index.js

```diff
@@ -7,116 +7,230 @@
 
 ================================================================================ */
 
-import { Client } from "@notionhq/client"
-import dotenv from "dotenv"
-import { Octokit } from "octokit"
+const { Client } = require("@notionhq/client")
+const dotenv = require("dotenv")
+const { Octokit } = require("octokit")
 
 dotenv.config()
 const octokit = new Octokit({ auth: process.env.GITHUB_KEY })
 const notion = new Client({ auth: process.env.NOTION_KEY })
 
-const database_id = process.env.NOTION_DATABASE_ID
+const databaseId = process.env.NOTION_DATABASE_ID
+const OPERATION_BATCH_SIZE = 10
 
-async function syncIssuesWithDatabase() {
-  console.log("Syncing GitHub Issues with Notion Database")
-  const issuesInDatabase = await getIssuesFromDatabase()
+/**
+ * Local map to store  GitHub issue ID to its Notion pageId.
+ * { [issueId: string]: string }
+ */
+const gitHubIssuesIdToNotionPageId = {}
 
-  // Get a list of github issues and add them to a local store.
-  let gitHubIssues = {}
+/**
+ * Initialize local data store.
+ * Then sync with GitHub.
+ */
+setInitialGitHubToNotionIdMap().then(syncNotionDatabaseWithGitHub)
+
+/**
+ * Get and set the initial data store with issues currently in the database.
+ */
+async function setInitialGitHubToNotionIdMap() {
+  const currentIssues = await getIssuesFromNotionDatabase()
+  for (const { pageId, issueNumber } of currentIssues) {
+    gitHubIssuesIdToNotionPageId[issueNumber] = pageId
+  }
+}
+
+async function syncNotionDatabaseWithGitHub() {
+  // Get all issues currently in the provided GitHub repository.
+  console.log("\nFetching issues from Notion DB...")
+  const issues = await getGitHubIssuesForRepository()
+  console.log(`Fetched ${issues.length} issues from GitHub repository.`)
+
+  // Group issues into those that need to be created or updated in the Notion database.
+  const { pagesToCreate, pagesToUpdate } = getNotionOperations(issues)
+
+  // Create pages for new issues.
+  console.log(`\n${pagesToCreate.length} new issues to add to Notion.`)
+  await createPages(pagesToCreate)
+
+  // Updates pages for existing issues.
+  console.log(`\n${pagesToUpdate.length} issues to update in Notion.`)
+  await updatePages(pagesToUpdate)
+
+  // Success!
+  console.log("\n✅ Notion database is synced with GitHub.")
+}
+
+/**
+ * Gets pages from the Notion database.
+ *
+ * @returns {Promise<Array<{ pageId: string, issueNumber: number }>>}
+ */
+async function getIssuesFromNotionDatabase() {
+  const pages = []
+  let cursor = undefined
+  while (true) {
```

Comment:
do/while no?

Reply:
I respectfully decline to use them for these scripts.